### PR TITLE
fix a variety of bugs in isolated support

### PR DESCRIPTION
**Pull request by ryanpetrello**

### CODE REVIEW

File: awx/main/expect/isolated_manager.py [hunk 1]
```diff
@@ -198,9 +198,20 @@ def check(self, interval=None):
...
                     if os.path.exists(path):
-                        event_data = json.load(
-                            open(os.path.join(events_path, event), 'r')
-                        )
+                        try:
+                            event_data = json.load(
+                                open(os.path.join(events_path, event), 'r')
+                            )
+                        except json.decoder.JSONDecodeError:
+                            # This means the event we got back isn't valid JSON
```
Comment (ryanpetrello) [inline awx/main/expect/isolated_manager.py]: @matburt before runner, the job events we wrote were written as partial files and *moved* into the final file name to provide atomic writes (so we avoided scenarios where we rsynced back files which were still being written).  I'm seeing evidence that this behavior is *not* preserved in runner:

image

...so I'm employing some defensive programming for now to work around this.

File: awx/main/tasks.py [hunk 2]
```diff
@@ -1187,14 +1187,17 @@ def run(self, pk, **kwargs):
             if self.instance.is_isolated() is True:
                 module_args = None
                 if 'module_args' in params:
+                    # if it's adhoc, copy the module args
                     module_args = ansible_runner.utils.args2cmdline(
                         params.get('module_args'),
                     )
+                else:
```
Comment (ryanpetrello) [inline awx/main/tasks.py:1194]: @chrismeyersfsu we only want to do this `copy_tree` call if it's a playbook

File: awx/playbooks/run_isolated.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@
         src: "{{src}}"
         dest: "{{dest}}"
 
-    - stat: path="{{src}}/env/ssh_key"
+    - local_action: stat path="{{src}}/env/ssh_key"
```
Comment (ryanpetrello) [inline awx/playbooks/run_isolated.yml:21]: @chrismeyersfsu this is why key copying wasn't working - we're checking to see if there *is* an SSH key to copy from the controller to the isolated node, but we were checking for file existence on the *isolated node*, **not** the controller. This faulty logic is my fault :)

**Review by AlanCoding — APPROVED** [review]
None of these changes look risky to me, and great job finding these bugs

**Pull request merged**